script.js

```diff
@@ -1,4 +1,339 @@
+      }
+      if (cardScore === 1) {
+        cardScore = 11;
+      }
+
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+      if (cardName == 1) {
+        cardName = "Ace";
+      } else if (cardName == 11) {
+        cardName = "Jack";
+      } else if (cardName == 12) {
+        cardName = "Queen";
+      } else if (cardName == 13) {
+        cardName = "King";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        score: cardScore,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cardDeck;
 };
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+/* things to do
+1) edit the card editor to add in points for blackjack score counting (D)
+3) plan for states
+  a) "initialize" should make it such that when you click submit. draws 2 cards, shows player the results and total score. tells you to hit or stand.
+  b) "playerMidgame" should have 2 inputs, hit or stand. plays a card if hit, shows the results and total score.
+   ace will be lowered to score 10 if a card is hit. check for bust, if bust and in the presence of ace, reduce ace to 1.
+   if stand, move game phase to "computerMidgame"
+  c) "computerMidgame" should basically make the cpu hit until the cpu is either bigger than player, or busts. then moves to "result" phase  
+  d) "result" phase shows who has won. moves it to cleaning phase
+  e) "scoreboard" phase will resets the deck, the hand and all required stuff on submit, moves it back to "initialize". also shows scoreboard
+*/
+
+let deck = shuffleCards(makeDeck());
+let gameState = "initialize";
+let playerHand = [];
+let computerHand = [];
+let playerScore = 0;
+let computerScore = 0;
+let playerBustStatus = false;
```

Comment:
Rather than having a global variable for the bust status, i would rather you create a helper function that takes in a hand and checks if it has bust. Reason being, this method is not scalable. Imagine having more than one player, you would need to have a variable for each user